turbopack/crates/turbopack-ecmascript/src/lib.rs

```diff
@@ -716,6 +732,51 @@ impl EcmascriptChunkPlaceable for EcmascriptModuleAsset {
     }
 }
 
+#[turbo_tasks::value_impl]
+impl MergeableModule for EcmascriptModuleAsset {
+    #[turbo_tasks::function]
+    async fn is_mergeable(self: ResolvedVc<Self>) -> Result<Vc<bool>> {
+        if matches!(
+            &*self.get_exports().await?,
+            EcmascriptExports::EsmExports(_)
+        ) {
+            return Ok(Vc::cell(true));
+        }
+
+        Ok(Vc::cell(false))
+    }
```

Comment:
this _might_ be a candidate for turning into a non-turbo-task trait item since you only call it once per module as part of the traversal code.  and it would be very unlikely to invalidate independently of the module graph itself.

Reply:
yeah and if you remove the last export from a module (or add the first export) you have almost certainly changed the module-graph so we will end up re-executing anyway, right?

Reply:
I was thinking like 

> Would direct trait usage help sync vs async problem, considering that it allows returning synchronously?

but `into_trait_ref` was returning a future